modules/imaging_browser/php/imaging_browser.class.inc

```diff
@@ -284,117 +98,84 @@ END";
-
-        $config = \NDB_Config::singleton();
-
-        $toTable_scan_types = $config->getSetting('tblScanTypes');
-
-        $all_scan_types_2d = $DB->pselect(
-            "SELECT ID, Scan_type FROM mri_scan_type mri",
-            array()
-        );
-
-        foreach ($all_scan_types_2d as $row) {
-            $type = $row['Scan_type'];
-            $all_scan_types[$row['ID']] = $type;
-        }
-
-        // Get the intersection between all the scan types and those
-        // that are desired to go into imaging browser table, based on
-        // array values rather than keys (hence the array_flip), then flip
-        // the resulting array back to revert it to a key/value (i.e.
-        // acquisition protocol ID/scan type) combination.
-
-        $scan_id_types          =array_flip(
-            array_intersect_key(
-                array_flip($all_scan_types),
-                array_flip($toTable_scan_types)
-            )
-        );
-        $modalities_subquery_as = array();
-        $as = array();
+        // get list of projects
         $list_of_projects = array();
-
-        foreach ($scan_id_types as $key => $value) {
-            $modalities_subquery[$key] = $value;
-            $as[$key] = $scan_id_types[$key] . '_QC_Status';
-            $modalities_subquery_as[$key] = $modalities_subquery[$key] .
-                ' as ' .
-                $as[$key];
-        }
-
-        $list_of_projects = $allAr;
         $projectList      = \Utility::getProjectList();
         foreach ($projectList as $key => $project) {
             unset($projectList[$key]);
             $list_of_projects[$project] = $project;
         }
 
+        // get visit QC status options
         $visitQCStatus = array(
                           'Pass' => 'Pass',
                           'Fail' => 'Fail',
                          );
 
+        // get Pending/New options
         $pending = array(
                     'P' => 'Pending',
                     'N' => 'New',
                    );
 
+        // get list of scan types
+        $DB         = \Database::singleton();
         $types_q    = $DB->pselect(
             "SELECT ID, Scan_type FROM mri_scan_type mri
                  JOIN files f ON (f.AcquisitionProtocolID=mri.ID)",
             array()
         );
-        $scan_types = $allAr;
+        $scan_types = array();
         foreach ($types_q as $key=>$sequenceType) {
             $scan_types[$sequenceType['Scan_type']] = $sequenceType['Scan_type'];
         }
 
-        $outputTypes = $DB->pselect(
-            "SELECT DISTINCT OutputType AS outputType 
-            FROM files WHERE FileType='mnc' AND OutputType!='native'",
+        // get list of scan types that should be present in the data table
+        $config             = \NDB_Config::singleton();
+        $toTable_scan_types = $config->getSetting('tblScanTypes');
+        $all_scan_types_2d  = $DB->pselect(
+            "SELECT ID, Scan_type FROM mri_scan_type mri",
```

Comment:
I have a feeling this is not right. It works though. @driusan is that OK to still have a query and use `$DB` in here?

Reply:
You need to get it from somewhere, but I think it would be a better idea to encapsulate it in some kind of `getScanTypes()` method (not sure if the proper place for that would be here or in the Utility class, but it would make the code easier to follow)

Reply:
OK. I will add it to the Utility class then as it might be useful for other modules down the road.